src/telemetry_registry.erl

```diff
@@ -114,13 +114,21 @@ gather_application_events(Application) ->
 
 -spec events_for_module(atom()) -> [telemetry:event_name() | event_definition()].
 events_for_module(Module) ->
-    Info = Module:module_info(),
-    Attributes = proplists:get_value(attributes, Info),
-    Events = proplists:get_all_values(telemetry_event, Attributes),
+    Attributes = case code:is_loaded(Module) of
+                     {file, _} -> Module:module_info(attributes);
+                     _ ->
+                         case code:get_object_code(Module) of
```

Comment:
Doing `code:which/1` and passing path to `beam_lib:chunks` should be more efficient since it allows avoiding slurping the whole file into memory and only reading the relevant parts

Reply:
I have "benchmarked" it (by running `time rebar3 ct`) and from my experiments it come out that `code:which/1` was slower than `code:get_object_code/1`.